pkg/logger/logger.go

```diff
@@ -0,0 +1,244 @@
+package logger
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Logger interface defines the minimal required logging methods
+type Logger interface {
+	Debug(msg string, fields ...Field)
+	Info(msg string, fields ...Field)
+	Warn(msg string, fields ...Field)
+	Error(msg string, fields ...Field)
+	WithContext(ctx context.Context) Logger
+	WithFields(fields ...Field) Logger
+	Sync() error
+}
+
+// Field represents a key-value pair for structured logging
+type Field struct {
+	Key   string
+	Value any
+}
+
+// Config holds minimal logger configuration
+type Config struct {
+	Level       string `json:"level"`       // "debug", "info", "warn", "error"
+	Development bool   `json:"development"` // true for development mode (console), false for production (json)
+}
+
+// zapLogger implements the Logger interface using uber/zap
+type zapLogger struct {
+	*zap.Logger
+}
+
+// DefaultConfig returns a default configuration for development
+func DefaultConfig() *Config {
+	return &Config{
+		Level:       "info",
+		Development: true,
+	}
+}
+
+// ProductionConfig returns a configuration for production
+func ProductionConfig() *Config {
+	return &Config{
+		Level:       "info",
```

Comment:
Are there any constants available from zap to specify levels?
I would prefer we use something like `zap.LEVEL_INFO` rather that a string `"info"`